generator.go

```diff
@@ -0,0 +1,71 @@
+//
+//  Author:: Salim Afiune <afiune@chef.io>
+//
+
+package main
+
+// This file will have the functions that will generate random data,
+// it involve creating fake Chef Nodes and Chef Runs that can be sent
+// to the data-collector endpoint
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/go-chef/chef"
+)
+
+func generateRandomData(nodeClient chef.Client, ohaiJSON, convergeJSON, complianceJSON map[string]interface{}) (err error) {
+	channels := make([]<-chan error, config.NumNodes)
+
+	for i := 0; i < config.NumNodes; i++ {
+		nodeName := config.NodeNamePrefix + "-" + strconv.Itoa(i+1)
```

Comment:
Right, chef-load currently has no way to keep track of what nodes have been created during previous chef-load runs. I’m not sure we want to start tracking that kind of state either but I’m open to suggestions. My suggestion is we can add something like a “nodeIndex” setting that lets us define what number chef-load starts counting from.

Reply:
I just added that to see different names but, let's keep in mind that Automate doesn't care about the node name, that is just a tag, what identifies a node as a node is the Entity UUID. Having said that, I don't know why, when we run it with 10 nodes twice, it doesn't generate new nodes, it continues with the same nodes but it adds extra chef-runs to the existing one. I can show that anytime with this PR: https://github.com/chef/a2/pull/390

Reply:
each node keeps it's UUID by design because that is what real infrastructure would do

a version 3 UUID is used to make this happen

https://github.com/chef/chef-load/blob/master/chef-client-run.go#L19

https://tools.ietf.org/html/rfc4122#section-4.3